src/redux/reducers/items.js

```diff
@@ -11,6 +11,13 @@ export default function(state = initialState, action) {
         [id]: itemFields
       };
     }
+    case ACTIONS.SELL_ITEM: {
+      const id = action.payload.id;
+      const changedItem = action.payload.item;
+      const newState = state;
```

Comment:
If you call this newState, this is where the object copy should be done.
Right now you are mutating global state before assigning ⚠️